BlogTemplate/Models/BlogDataStore.cs

```diff
@@ -20,13 +20,44 @@ public BlogDataStore(IFileSystem fileSystem)
         {
             _fileSystem = fileSystem;
             InitStorageFolder();
+            InitCurrentId();
         }
         public void InitStorageFolder()
         {
             _fileSystem.CreateDirectory(StorageFolder);
             _fileSystem.CreateDirectory(DraftsFolder);
         }
 
+        public void InitCurrentId()
+        {
+            int max = 0;
+            List<string> postfiles = _fileSystem.EnumerateFiles($"{StorageFolder}").ToList();
+            foreach (var file in postfiles)
+            {
+                int start = file.LastIndexOf("\\");
+                int end = file.IndexOf(".");
+                int currId = Convert.ToInt32(file.Substring(start + 1, end - start - 1));
+                if(currId > max)
+                {
+                    max = currId;
+                }
+            }
+
+            List<string> draftfiles = _fileSystem.EnumerateFiles($"{DraftsFolder}").ToList();
+            foreach(var file in draftfiles)
+            {
+                int start = file.IndexOf("_");
+                int end = file.IndexOf(".");
+                int currId = Convert.ToInt32(file.Substring(start + 1, end - start - 1));
+                if (currId > max)
+                {
+                    max = currId;
+                }
+            }
+
+            CurrentId = max;
+        }
+
         public void SetId(Post post)
         {
             lock(thisLock)

```

Comment:
You could also consider naming this something like EnsureValidId().  And it should be private.